connector/src/main/java/com/google/cloud/hive/bigquery/connector/BigQueryStorageHandler.java

```diff
@@ -146,6 +156,37 @@ public void configureOutputJobProperties(TableDesc tableDesc, Map<String, String
       ProtoSchema protoSchema =
           com.google.cloud.bigquery.storage.v1.ProtoSchemaConverter.convert(descriptor);
       jobInfo.setProtoSchema(protoSchema.toByteArray());
+    } else if (writeMethod.equals(HiveBigQueryConfig.WRITE_METHOD_INDIRECT)) {
+      // Figure out the table's Avro schema
+      org.apache.avro.Schema avroSchema;
+      boolean hasExternalSchema =
+          tableProperties.getProperty(
+                      AvroSerdeUtils.AvroTableProperties.SCHEMA_LITERAL.getPropName())
+                  != null
+              || tableProperties.getProperty(
+                      AvroSerdeUtils.AvroTableProperties.SCHEMA_URL.getPropName())
+                  != null;
+      if (!hasExternalSchema
```

Comment:
Can we extract this if logic out in a separate method that returns boolean to increase readability?

We could also create a separate method for the boolean variable hasExternalSchema as well

Reply:
Yes good idea, moving the code the a separate method.